### PR TITLE
Add new event before a customer is actually registered

### PR DESCRIPTION
**Pull request by t2oh4e**
### 1. Why is this change necessary?
Because there is currently no clean way to abort a registration process beside the form validation. You can decorate the RegisterService but since this does not return anything you can not pass back any information to the register controller to initiate a custom handling. I had the need to do this because I wanted to abort registrations after the form validation is run but before the customer is actually saved.

### 2. What does this change do, exactly?
Add a new event to handle custom errors (or messages in general) during the registration process.

### 3. Describe each step to reproduce the issue or behaviour.
Try to abort a registration before a customer is saved to the database.
Example subscriber that could make use of the new event:
```php
class Test implements SubscriberInterface {
    public static function getSubscribedEvents()
    {
        return [
            'Shopware_Modules_Admin_SaveRegister_BeforeRegister' => 'onBeforeRegister',
        ];
    }

    public function onBeforeRegister(\Enlight_Event_EventArgs $args) {
        $errors = $args->getReturn();
        $customer = $args->get('customer');
        $billing = $args->get('billing');
        $shipping = $args->get('shipping');

        // Do some custom validation stuff with the customer or the addresses
        // e.g. sync them with another system and react to API errors of that system.

        // Merge to make sure you keep errors that may have been added by other plugins previously.
        return array_merge($errors, [
           'occurred' => true,
           'personal' => [
               //reuse error structure from register controller which will be displayed in the frontend form
               'firstname' => 'Something is wrong here',
           ],
           'my_custom_errors' => [
               // Add your own errors. They will be passed to the view and you can handle them there.
           ]
        ]);
    }
}
```

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Controllers/Frontend/Register.php [hunk 1]
```diff
@@ -185,6 +177,26 @@ public function saveRegisterAction()
...
 
+        /** @var Enlight_Event_EventManager $eventManager */
+        $eventManager = $this->get('events');
+
+        $errors = ['occurred' => false];
+        $errors = $eventManager->filter(
+            'Shopware_Modules_Admin_SaveRegister_BeforeRegister',
+            $errors,
+            [
+                'customer' => $customer,
```
Comment (shyim) [inline engine/Shopware/Controllers/Frontend/Register.php:189]: subject missing

Comment (t2oh4e) [reply engine/Shopware/Controllers/Frontend/Register.php:189]: Added.

**Comment by t2oh4e** [comment]
@shopwareBot Sorry I was on vacations :) I added the missing subject requested by shyim.

**Comment by mitelg** [comment]
shouldn't we avoid to add `subject` to events? so in case of refactoring, it would be easier to move the code :wink:

**Comment by t2oh4e** [comment]
@mitelg sounds legit. I found both ways (with and without subject) in the existing code. I can remove it again. Shall I?

**Comment by shyim** [comment]
Thanks for your Contribution! :blue_heart:

**Pull request closed without merging**